src/main/java/org/sagebionetworks/web/server/servlet/AppConfigServlet.java

```diff
@@ -0,0 +1,155 @@
+  ) {
+    this.appConfigDataClient = appConfigDataClient;
+    this.stackConfiguration = stackConfiguration;
+  }
+
+  @Override
+  public void init() throws ServletException {
+    super.init();
+    initializeAppConfigClient();
+  }
+
+  public void initializeAppConfigClient() {
+    if (appConfigDataClient == null) {
+      appConfigDataClient = AwsClientFactory.createAppConfigClient();
+    }
+    startConfigurationSession();
+    initializeConfigSupplier();
+  }
+
+  public void startConfigurationSession() {
+    try {
+      StartConfigurationSessionRequest sessionRequest =
+        new StartConfigurationSessionRequest()
+          .withApplicationIdentifier(
+            stackConfiguration.getStack() +
+            "-" +
+            stackConfiguration.getStackInstance() +
+            "-portal-AppConfigApp"
+          )
+          .withEnvironmentIdentifier(
+            stackConfiguration.getStack() +
+            "-" +
+            stackConfiguration.getStackInstance() +
+            "-portal-environment"
+          )
+          .withConfigurationProfileIdentifier(
+            stackConfiguration.getStack() +
+            "-" +
+            stackConfiguration.getStackInstance() +
+            "-portal-configurations"
+          );
+      StartConfigurationSessionResult sessionResponse =
+        appConfigDataClient.startConfigurationSession(sessionRequest);
+      configurationToken = sessionResponse.getInitialConfigurationToken();
+    } catch (Exception e) {
+      configurationToken = null;
+    }
+  }
+
+  public void initializeConfigSupplier() {
+    configSupplier =
+      Suppliers.memoizeWithExpiration(
+        this::getLatestConfiguration,
+        5,
+        TimeUnit.MINUTES
+      );
+  }
+
+  public String getLatestConfiguration() {
+    if (configurationToken == null) {
+      return defaultConfigValue;
+    }
+    try {
+      GetLatestConfigurationRequest latestConfigRequest =
+        new GetLatestConfigurationRequest()
+          .withConfigurationToken(configurationToken);
+      GetLatestConfigurationResult latestConfigResponse =
+        appConfigDataClient.getLatestConfiguration(latestConfigRequest);
+      configurationToken = latestConfigResponse.getNextPollConfigurationToken();
+      ByteBuffer configData = latestConfigResponse.getConfiguration();
+      String newConfigValue = new String(
+        configData.array(),
+        java.nio.charset.StandardCharsets.UTF_8
+      ); // This may be empty if the client already has the latest version of the configuration.
+      if (!newConfigValue.isEmpty()) {
+        lastConfigValue = newConfigValue;
+      }
+      return lastConfigValue;
+    } catch (Exception e) {
+      return defaultConfigValue;
+    }
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
+    throws ServletException, IOException {
+    try {
+      String configValue = configSupplier.get();
+      response.setContentType("text/plain");
```

Comment:
We are always serving JSON, right?
```suggestion
      response.setContentType("application/json");
```